app/celer-geo/Runner.cc

```diff
@@ -0,0 +1,228 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celer-geo/Runner.cc
+//---------------------------------------------------------------------------//
+#include "Runner.hh"
+
+#include "celeritas_config.h"
+#include "corecel/io/StringUtils.hh"
+#include "corecel/sys/Device.hh"
+#include "corecel/sys/Stopwatch.hh"
+#include "geocel/rasterize/RaytraceImager.hh"
+#include "orange/OrangeParams.hh"
+#if CELERITAS_USE_GEANT4
+#    include "geocel/g4/GeantGeoParams.hh"
+#endif
+#if CELERITAS_USE_VECGEOM
+#    include "geocel/vg/VecgeomParams.hh"
+#endif
+
+namespace celeritas
+{
+namespace app
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with model setup.
+ */
+Runner::Runner(ModelSetup const& input) : input_{input}
+{
+    // Initialize GPU
+    activate_device();
+
+    if (input_.cuda_heap_size)
+    {
+        set_cuda_heap_size(input_.cuda_heap_size);
+    }
+    if (input_.cuda_stack_size)
+    {
+        set_cuda_stack_size(input_.cuda_stack_size);
+    }
+
+    if (CELERITAS_USE_GEANT4 && ends_with(input_.geometry_file, ".gdml"))
+    {
+        // Retain the Geant4 world for possible reuse across geometries
+        this->load_geometry<Geometry::geant4>();
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Perform a raytrace.
+ */
+auto Runner::operator()(TraceSetup const& trace, ImageInput const& image_inp)
+    -> SPImage
+{
+    // Create image params
+    last_image_ = std::make_shared<ImageParams>(image_inp);
+
+    return (*this)(trace);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Perform a raytrace using the last image but a new geometry/memspace.
```

Comment:
I understand repeating in a different memspace (execution space?) but is there an actual use case for repeating the same trace with a new geometry?

Reply:
Oh absolutely, that's going to be the most common use case for the immediate future: it's the cleanest way to do geometry validation, so it lets us compare ORANGE rendering versus Geant4 versus VecGeom. It would've been really nice to have three years ago when VecGeom was riddled with bugs...